internal/pkg/githubapi/pr_metrics.go

```diff
@@ -0,0 +1,108 @@
+package githubapi
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/go-github/v62/github"
+	lru "github.com/hashicorp/golang-lru/v2"
+	log "github.com/sirupsen/logrus"
+	prom "github.com/wayfair-incubator/telefonistka/internal/pkg/prometheus"
+)
+
+const (
+	timeToDefineStale = 20 * time.Minute
+	metricRefreshTime = 60 * time.Second
+)
+
+func MainGhMetricsLoop(mainGhClientCache *lru.Cache[string, GhClientPair]) {
+	for t := range time.Tick(metricRefreshTime) {
+		log.Debugf("Updating pr metrics at %v", t)
+		getPrMetrics(mainGhClientCache)
+	}
+}
+
+func getRepoPrMetrics(ctx context.Context, ghClient GhClientPair, repo *github.Repository) (prWithStaleChecks int, openPRs int, openPromotionPrs int, err error) {
+	log.Debugf("Checking repo %s", repo.GetName())
+	ghOwner := repo.GetOwner().GetLogin()
+	prListOpts := &github.PullRequestListOptions{
+		State: "open",
+	}
+	prs := []*github.PullRequest{}
+
+	// paginate through PRs, there might be lots of them.
+	for {
+		perPagePrs, resp, err := ghClient.v3Client.PullRequests.List(ctx, ghOwner, repo.GetName(), prListOpts)
+		_ = prom.InstrumentGhCall(resp)
+		if err != nil {
+			log.Errorf("error getting PRs for %s/%s: %v", ghOwner, repo.GetName(), err)
+		}
```

Comment:
Why not checking the error first, and then calling the `prom.InstrumentGhCall(resp)`?
Another question if the output for  `prom.InstrumentGhCall(resp)` is not important why do bother to assign it something?

Reply:
The `InstrumentGhCall` function returns labels just for testing.
During runtime these labels are just used internally in the function.
I chose to instrument before error handling to ensure we still instrument even if we change the error handling to break out of that code path in the future, be I can change the order if you feel it makes it less readable

Reply:
>  I chose to instrument before error handling to ensure we still instrument even if we change the error handling to break out of that code path in the future, be I can change the order if you feel it makes it less readable

I would add that as a comment